external/fv3fit/fv3fit/pytorch/graph.py

```diff
@@ -0,0 +1,183 @@
+import torch
+import tensorflow as tf
+import numpy as np
+import dgl
+import dataclasses
+from .._shared.training_config import Hyperparameters
+from toolz.functoolz import curry
+from fv3fit.pytorch.graph_predict import PytorchModel
+from fv3fit.pytorch.graph_builder import build_graph, GraphConfig
+from fv3fit.pytorch.graph_config import GraphNetwork, GraphNetworkConfig
+from fv3fit.pytorch.graph_loss import LossConfig
+from fv3fit.pytorch.graph_optim import OptimizerConfig
+from fv3fit.pytorch.training_loop import TrainingLoopConfig
+from fv3fit._shared.scaler import StandardScaler
+
+from fv3fit._shared import register_training_function
+from typing import (
+    List,
+    Optional,
+    Sequence,
+    Set,
+    Mapping,
+)
+from fv3fit.tfdataset import select_keys, ensure_nd, apply_to_mapping
+
+
+@dataclasses.dataclass
+class GraphHyperparameters(Hyperparameters):
+    """
+    Args:
+        input_variables: names of variables to use as inputs
+        output_variables: names of variables to use as outputs
+        optimizer_config: selection of algorithm to be used in gradient descent
+        graph_network: configuration of graph network
+        training_loop: configuration of training loop
+        loss: configuration of loss functions, will be applied separately to
+            each output variable
+    """
+
+    input_variables: List[str]
+    output_variables: List[str]
+    normalization_fit_samples: int = 50_000
+    optimizer_config: OptimizerConfig = dataclasses.field(
+        default_factory=lambda: OptimizerConfig("AdamW")
+    )
+    build_graph: GraphConfig = dataclasses.field(default_factory=lambda: GraphConfig())
```

Comment:
Please rename this attribute to a noun rather than a verb. `graph_config`? Just `graph`?